napari/components/layerlist.py

```diff
@@ -62,15 +62,16 @@ def _coerce_name(self, name, layer=None):
         new_name : str
             Coerced, unique name.
         """
+        existing_layers = {x.name for x in self}
         if layer is None:
-            for existing_name in sorted(x.name for x in self):
-                if name == existing_name:
+            for i in range(len(self)):
+                if name in existing_layers:
                     name = inc_name_count(name)
         else:
             for _layer in sorted(self, key=lambda x: x.name):
                 if _layer is layer:
                     continue
-                if name == _layer.name:
+                if name in existing_layers:
                     name = inc_name_count(name)
         return name
```

Comment:
napari does no properly work with auto numerate above 10 layers.  `sorted` works lexicographical, not in human natural order.

Reply:
good catch!